crates/context/config/src/client.rs

```diff
@@ -63,199 +64,237 @@ impl Client<AnyTransport> {
         });
 
         let local = Self::from_local_config(&config).expect("validation error");
-
         let transport = transport!(local.transport, relayer);
-
         Self::new(transport)
     }
 
     pub fn from_local_config(config: &ClientConfig) -> eyre::Result<Client<LocalTransports>> {
         let mut near_transport = None;
+        let mut starknet_transport = None;
+        let mut ethereum_transport = None;
```

Comment:
Use a map here instead of copy/paste transports and use the network type as key?